### PR TITLE
Wait for thrift port to open before connecting

### PR DESCRIPTION
**Pull request by ptnapoleon**

### CODE REVIEW

**Comment by jkni** [comment]
Makes sense to me - we start the Native Server before the Thrift Server, so that could definitely race.

File: thrift_tests.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def post_initialize_cluster(cls):
             node1.set_configuration_options(values={'initial_token': "a".encode('hex')})
 
         cluster.start(wait_for_binary_proto=True)
+        cluster.nodelist()[0].watch_log_for("Listening for thrift clients")  # Wait for the thrift port to open
         time.sleep(0.1)
```
Comment (mambocab) [inline thrift_tests.py]: Will we still need this sleep?

Comment (jkni) [inline thrift_tests.py]: Don't think so - it was probably an attempt to work around this issue.

Comment (ptnapoleon) [inline thrift_tests.py]: I imagine so? I have documentation saying that the log message "Listening for X clients" actually occurs slightly before the action. Now, I suppose I could go look at the source, but the `0.1s` isn't hurting anyone?

**Comment by ptnapoleon** [comment]
Merging with the "lgtm" from joel, and an offline "+1" from Hatch.

**Comment by knifewine** [comment]
belated +1

**Comment by ptnapoleon** [comment]
Turns out Russ didn't give me a +1, and I just can't read offline messages properly. Alas.

**Pull request merged**